### ISSUE TITLE
Add 'admin' user tag in Auth0, pull through to client

### ISSUE THREAD

**Issue by cawarren** [Recidiviz/covid19-dashboard#183]
**Is your feature request related to a problem? Please describe.**
We provide a daily briefing to states which request one today, but through a fairly arduous manual data collection process (for them and for us).

The web tool allows states to maintain a baseline scenario (reflecting their current circumstances) that provides the information needed for the daily briefing, and in a fairly simple to maintain fashion. However, at the moment we have no easy way for our team to get access to this data for production of the daily briefings.

**Describe the solution you'd like**
1) create a rule within Auth0 to explicitly pass user metadata back in auth requests, and 
2) have functions in the app that can pull the admin tag out of the metadata and gate certain pages based on that tag.

**Additional context**
Child issue of issue #170

**Comment by macfarlandian** [comment]
as I dig into this I have a question about the implementation (mainly for @jessex I think) ... when we were putting together the sharing feature we decided that Auth0 would be used only for authentication, not authorization. With the approach laid out here, that would no longer be the case. I don't necessarily have a problem with that (having two sources of authorization is not the _best_ thing but it's not the end of the world) but I'm just wondering what the rationale was, or if it's something we should rethink now if it wasn't really a concern when this ticket was originally written? (the most likely alternative would be some kind of access control list maintained in Firestore)

**Comment by jessex** [comment]
Fair points! In my mind, this question doesn't turn on the concept of _who performs authorization_ but _what are we authenticating/authorizing_. In the case of sharing, we wanted to authorize access to specific scenarios, specifically we wanted to add complexity to already standing functionality for authorizing access to specific scenarios. And in that case, the scenarios and related entities were things that lived in Firestore and there was no good technical justification to move all of the scenario authorization work out of Firestore and into Auth0.

In the case of having admin pages, we are more directly authorizing the _user itself_, saying that if this is a certain _kind_ of user then they can access a certain page. The user entity and their metadata is something that essentially lives in Auth0, and nothing that is in Firestore cares _directly_ about whether a user is an admin or not.

If you think I'm underthinking this or missing something, let me know. It's certainly true that in the sharing work we copied user documents into Firestore for ease of authorization of scenarios, and that the case could be made that the admin flags should live directly with those documents in Firestore. But then again, I could foresee of future authentication/authorization requirements related to is_admin or other user-type flags, and as such cases expand we would find that we need this information in Auth0.

**Comment by macfarlandian** [comment]
no, I think you're right ... I actually think adding these flags into the user documents in Firestore would be a bad idea given the trouble we had with properly securing that collection, so we would have to come up with some other, more cumbersome solution anyway. I was on the fence and needed a sanity check but now I'm convinced this is the right way to go. Thanks!
